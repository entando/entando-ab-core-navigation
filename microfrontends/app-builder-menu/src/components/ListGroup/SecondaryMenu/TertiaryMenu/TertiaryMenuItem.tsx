import { useContext } from 'react';
import styled from 'styled-components';
import MenuUIContext from '../../../MenuUIContext';

const StyledLink = styled.a<{ isActive: boolean }>`
  width: 100%;
  background-color: transparent;
  background-color: ${({ isActive }) => (isActive ? '#4D5257' : 'transparent')};
  color: ${({ isActive }) => (isActive ? '#fff' : '#d1d1d1')};
  cursor: pointer;
  display: block;
  font-size: 14px;
  font-weight: ${({ isActive }) => (isActive ? '600' : '400')};
  line-height: 26px;
  padding: 5px 10px 5px 8px;
  position: relative;
  white-space: nowrap;
  text-decoration: none;
  display: flex;
  margin-top: 4px;

  &:hover {
    font-weight: 600;
    color: #fff;
    background-color: #4d5257;
  }
`;

const StyledLabel = styled.span`
  display: block;
  flex: 1;
  max-width: none;
  overflow: hidden;
  text-overflow: ellipsis;
`;

interface Props {
  id: string;
  label: string;
  onClick?: () => void;
}

export default function TertiaryMenuItem(props: Props): JSX.Element {
  const { id, label, onClick } = props;

  const { activeTertiaryMenuItemId, setActiveTertiaryMenuItemId } =
    useContext(MenuUIContext);

  const isActive = activeTertiaryMenuItemId === id;

  const handleClick = (): void => {
    setActiveTertiaryMenuItemId(id);
    onClick && onClick();
  };

  return (
    <StyledLink isActive={isActive} onClick={handleClick}>
      <StyledLabel>{label}</StyledLabel>
    </StyledLink>
  );
}
