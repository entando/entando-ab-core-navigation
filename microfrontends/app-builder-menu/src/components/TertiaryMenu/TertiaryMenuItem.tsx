import { useContext } from 'react';
import styled from 'styled-components';
import { MenuUIContext } from './../MenuUIContext';
import { COLORS } from './../theme';

const StyledLink = styled.a<{ isActive: boolean }>`
  width: 100%;
  background-color: transparent;
  background-color: ${({ isActive }) =>
    isActive ? COLORS.secondary : 'transparent'};
  color: ${({ isActive }) => (isActive ? COLORS.white : COLORS.text)};
  cursor: pointer;
  display: block;
  font-size: 14px;
  font-family: 'Open Sans', sans-serif;
  font-weight: ${({ isActive }) => (isActive ? '600' : '400')};
  line-height: 26px;
  padding: 5px 10px 5px 8px;
  position: relative;
  white-space: nowrap;
  text-decoration: none;
  display: flex;
  margin-top: 4px;

  &:hover {
    font-weight: 600;
    color: ${COLORS.white};
    background-color: ${COLORS.secondary};
  }
`;

const StyledLabel = styled.span`
  display: block;
  flex: 1;
  max-width: none;
  overflow: hidden;
  text-overflow: ellipsis;
`;

interface Props {
  id: string;
  dataId?: string;
  label: string;
  onClick?: () => void;
  href?: string;
  className?: string;
  target?: string;
  rel?: string;
}

export function TertiaryMenuItem(props: Props): JSX.Element {
  const {
    id,
    dataId,
    label,
    onClick,
    href,
    target,
    rel,
    className = ''
  } = props;

  const {
    activeTertiaryMenuItemId,
    setActiveTertiaryMenuItemId,
    setTertiaryMenuOpen
  } = useContext(MenuUIContext);

  const isActive = activeTertiaryMenuItemId === id;

  const handleClick = (): void => {
    setActiveTertiaryMenuItemId(id);
    setTertiaryMenuOpen(false);
    onClick && onClick();
  };

  return (
    <StyledLink
      isActive={isActive}
      onClick={handleClick}
      href={href}
      className={className}
      rel={rel}
      target={target}
      data-id={dataId}
    >
      <StyledLabel>{label}</StyledLabel>
    </StyledLink>
  );
}
