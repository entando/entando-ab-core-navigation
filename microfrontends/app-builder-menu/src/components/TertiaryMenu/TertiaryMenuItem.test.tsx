import React from 'react';
import renderer from 'react-test-renderer';

import { TertiaryMenuItem } from './TertiaryMenuItem';

test('renders TertiaryMenuItem without href', () => {
  const tree = renderer
    .create(<TertiaryMenuItem id="item1" label="Item 1" onClick={() => {}} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test('renders TertiaryMenuItem with href, target and rel', () => {
  const tree = renderer
    .create(
      <TertiaryMenuItem
        id="item1"
        label="Item 1"
        target="_blank"
        rel="noopener noreferrer"
        href="http://localhost:8080"
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
