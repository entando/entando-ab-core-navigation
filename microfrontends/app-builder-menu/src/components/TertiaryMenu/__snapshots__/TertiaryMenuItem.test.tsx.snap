// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders TertiaryMenuItem with href, target and rel 1`] = `
<a
  className="sc-bczRLJ chTIJh"
  href="http://localhost:8080"
  onClick={[Function]}
  rel="noopener noreferrer"
  target="_blank"
>
  <span
    className="sc-gsnTZi DkFNS"
  >
    Item 1
  </span>
</a>
`;

exports[`renders TertiaryMenuItem without href 1`] = `
<a
  className="sc-bczRLJ chTIJh"
  onClick={[Function]}
>
  <span
    className="sc-gsnTZi DkFNS"
  >
    Item 1
  </span>
</a>
`;
